src/main/java/mate.academy/service/impl/ProductServiceImpl.java

```diff
@@ -2,13 +2,18 @@
 
 import java.util.List;
 import java.util.stream.Collectors;
+import mate.academy.lib.Component;
+import mate.academy.lib.Inject;
 import mate.academy.model.Product;
 import mate.academy.service.FileReaderService;
 import mate.academy.service.ProductParser;
 import mate.academy.service.ProductService;
 
+@Component
 public class ProductServiceImpl implements ProductService {
+    @Inject
     private ProductParser productParser;
+    @Inject
     private FileReaderService fileReaderService;
```

Comment:
Consider making the Interface Implementations map a class field and fill it in using `Map.of()`. This will improve the structure of your dependency injection mechanism by centralizing the interface-to-implementation mapping.